### ISSUE TITLE
Regression for `float_cmp` in `assert_eq!(x, 1.0)`

### ISSUE THREAD

**Issue by George-Ogden** [rust-lang/rust-clippy#17570]
### Description

This code snippet no longer raises any diagnostic: `assert_eq!(x, 1.0)`
It should raise `float_cmp` or `float_cmp_const`.
The regression was introduced in #17024

### Version

```text
rustc 1.99.0-nightly (84b36a78a 2026-08-06)
binary: rustc
commit-hash: 84b36a78a28a63f134171c670be1932ffa2485f8
commit-date: 2026-08-06
host: x86_64-unknown-linux-gnu
release: 1.99.0-nightly
LLVM version: 23.1.0
```
